reference/src/taimpl/src/porting/overflow.c

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright 2023 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "porting/overflow.h"
+
+bool add_overflow(
+        unsigned long a,
+        unsigned long b,
+        unsigned long* result) {
+    return __builtin_uaddl_overflow(a, b, result);
```

Comment:
while I highly doubt anyone is gonna use a non-gcc compiler, do you still want to put like an `#ifdef __GNUC__` in case some vendor specific soc isn't using GCC?

Reply:
This is in the porting directory. So I want the SOC vendor fix the compile error if this function isn't provided. Incidentally, Apple's compiler does provide it even though it's not a GCC compiler.